scripts/zephyr_module.py

```diff
@@ -394,9 +395,13 @@ def west_projects():
     return None
 
 
-def parse_modules(zephyr_base, modules=None, extra_modules=None):
+def parse_modules(zephyr_base, manifest=None, west_projs=None, modules=None,
+                  extra_modules=None):
+
     if modules is None:
-        modules = []
+        west_projs = west_projs or west_projects(manifest)
+        modules = ([p.posixpath for p in west_projs['projects']]
+                   if modules else [])
```

Comment:
you just tested for `modules` 3 lines above, so modules is for sure `None` meaning we always get `[]` here.

Reply:
Yep, this was a mistake I introduced, thanks!